src/briefcase/integrations/java.py

```diff
@@ -26,28 +30,40 @@ def __init__(self, tools: ToolCache, java_home: Path):
 
     @property
     def OpenJDK_download_url(self):
-        arch = {
-            "x86_64": "x64",  # Linux\macOS x86-64
-            "aarch64": "aarch64",  # Linux arm64
-            "armv6l": "arm",  # Linux arm
-            "arm64": "aarch64",  # macOS arm64
-            "AMD64": "x64",  # Windows x86-64
-        }.get(self.tools.host_arch)
-
-        platform = {
-            "Darwin": "mac",
-            "Windows": "windows",
-            "Linux": "linux",
-        }.get(self.tools.host_os)
-
-        extension = {
-            "Windows": "zip",
-        }.get(self.tools.host_os, "tar.gz")
+        """The OpenJDK download URL appropriate for the current machine."""
+        system_arch = self.tools.host_arch
+        # use a 32bit JDK if using 32bit Python on 64bit hardware
+        if self.tools.is_32bit_python and self.tools.host_arch == "aarch64":
+            system_arch = "armv7l"
```

Comment:
Downloading a 32bit JDK on ARM when using a 32bit Python is mostly a proxy for assuming the OS installed is also 32bit.